models/superrare/superrare_ethereum_events.sql

```diff
@@ -0,0 +1,372 @@
+    WHERE evt_block_time >= date_trunc("day", now() - interval '1 week')
+    {% endif %}
+
+    union all 
+    
+    select  block_time
+            , concat('0x',substring(topic2 from 27 for 40)) as contract_address 
+            , bytea2numeric_v2(substring(topic4 from 3)) as token_id 
+            , lower('0x8c9f364bf7a56ed058fc63ef81c6cf09c833e656') as seller -- all sent from auction house contract 
+            , concat('0x',substring(topic3 from 27 for 40)) as buyer 
+            , bytea2numeric_v2(substring(data from 67 for 64)) as amount
+            , tx_hash
+            , '' as `_currencyAddress`
+    -- from ethereum.logs
+    from {{ source('ethereum','logs') }}
+    where contract_address = lower('0x8c9f364bf7a56ed058fc63ef81c6cf09c833e656')
+        and topic1 = lower('0xea6d16c6bfcad11577aef5cc6728231c9f069ac78393828f8ca96847405902a9')
+    {% if is_incremental() %} -- this filter will only be applied on an incremental run
+    and block_time >= date_trunc("day", now() - interval '1 week')
+    {% endif %}
+
+    union all 
+    
+    select block_time
+            , concat('0x',substring(topic2 from 27 for 40)) as contract_address 
+            , bytea2numeric_v2(substring(data from 67 for 64)) as token_id
+            , concat('0x',substring(topic4 from 27 for 40)) as seller 
+            , concat('0x',substring(topic3 from 27 for 40)) as buyer 
+            , bytea2numeric_v2(substring(data from 3 for 64)) as amount
+            , tx_hash
+            , '' as `_currencyAddress`
+    -- from ethereum.logs
+    from {{ source('ethereum','logs') }}
+    where contract_address = lower('0x65b49f7aee40347f5a90b714be4ef086f3fe5e2c')
+        and topic1 in (lower('0x2a9d06eec42acd217a17785dbec90b8b4f01a93ecd8c127edd36bfccf239f8b6')
+                        , lower('0x5764dbcef91eb6f946584f4ea671217c686fa7e858ce4f9f42d08422b86556a9'))
+    {% if is_incremental() %} -- this filter will only be applied on an incremental run
+    and block_time >= date_trunc("day", now() - interval '1 week')
+    {% endif %}
+
+)
+
+-- some items are sold in RARE currency on superrare. not available on coinpaprika, so using dex data to be able to convert to USD. usuing weekly average since dex data isn't fully populated on v2 right now (~1/8 of data vs. v1). switch back to daily once full data is available
+, rare_token_price_eth as ( 
+    with all_rare_eth_dex_trades as (
+        select  block_time
+                , token_bought_symbol
+                , token_sold_symbol
+                , token_bought_amount
+                , token_sold_amount 
+                , case when token_bought_symbol like '%ETH%' then token_bought_amount*1.0/nullif(token_sold_amount,0) 
+                    else  token_sold_amount*1.0/nullif(token_bought_amount,0)
+                end as eth_per_rare
+        -- from dex.trades
+        from {{ ref('dex_trades') }}
+        where   -- RARE trades
```

Comment:
`{% if is_incremental() %} -- this filter will only be applied on an incremental run
        where block_time >= date_trunc("day", now() - interval '1 week')
        {% endif %}
        and   -- RARE trades`